src/WebJobs.Extensions.OpenAI/Assistants/AssistantService.cs

```diff
@@ -63,11 +63,14 @@ public async Task CreateAssistantAsync(AssistantCreateRequest request, Cancellat
             request.Id,
             request.Instructions ?? "(none)");
 
-        this.CreateTableClient(request);
+        if (this.tableClient is null)
+        {
+            this.CreateTableClient(request.ChatStorageConnectionSetting, request.CollectionName);
+        }
 
         if (this.tableClient is null)
         {
-            throw new ArgumentNullException(nameof(this.tableClient));
+            throw new InvalidOperationException($"{nameof(this.tableClient)} is null.");
         }
 
         // Create the table if it doesn't exist
```

Comment:
the next line calls CreateTableClientIfNotExists. Do we need that anymore?

Reply:
next line is about creating table if not exists.

table client creation and table creation are separate, can't change the method name as it is Azure Tables library.